core/client/db/src/lib.rs

```diff
@@ -2134,18 +2177,58 @@ mod tests {
 		let block1 = insert_header(&backend, 1, block0, Vec::new(), Default::default());
 
 		{
-			let tree_route = ::client::blockchain::tree_route(
-				|id| blockchain.header(id)?.ok_or_else(|| client::error::Error::UnknownBlock(format!("{:?}", id))),
-				BlockId::Hash(block0),
-				BlockId::Hash(block1),
-			).unwrap();
+			let tree_route = tree_route(&blockchain, block0, block1).unwrap();
 
 			assert_eq!(tree_route.common_block().hash, block0);
 			assert!(tree_route.retracted().is_empty());
 			assert_eq!(tree_route.enacted().iter().map(|r| r.hash).collect::<Vec<_>>(), vec![block1]);
 		}
 	}
 
+	#[test]
+	fn lowest_common_ancestor_works() {
```

Comment:
Is this test actually using ancestor links anywhere? Maybe we could add some more edge cases.

Reply:
It's is used because they are executed before and initialized to parents. I added a couple of cases more.